crates/storage/store.rs

```diff
@@ -358,6 +359,57 @@ impl Store {
         Ok(Some(state_trie.hash()?))
     }
 
+    /// Applies state transitions to the given trie and does not commit nor calculate root hash
+    /// Note: it does commit storage tries and new account codes
+    pub fn apply_account_updates_to_trie(
```

Comment:
this seems to me that it should be more of a utility function inside `blockchain`, thoughts?